pkg/integrationteststatus/integration_test_status.go

```diff
@@ -163,18 +166,24 @@ func (sits *SnapshotIntegrationTestStatuses) UpdateTestStatusIfChanged(scenarioN
 
 }
 
-// UpdatePipelineRunName updates TestPipelineRunName if changed
+// UpdateTestPipelineRunName updates TestPipelineRunName if changed
 // scenario must already exist in statuses
 func (sits *SnapshotIntegrationTestStatuses) UpdateTestPipelineRunName(scenarioName string, pipelineRunName string) error {
 	detail, ok := sits.GetScenarioStatus(scenarioName)
 	if !ok {
-		return fmt.Errorf("scenario name %s not found and cannot be updated", scenarioName)
+		return fmt.Errorf("scenario name %s not found within the SnapshotIntegrationTestStatus, and cannot be updated", scenarioName)
 	}
 
 	if detail.TestPipelineRunName != pipelineRunName {
 		detail.TestPipelineRunName = pipelineRunName
 		sits.dirty = true
 	}
+
+	if detail.Status == IntegrationTestStatusCancelled {
+		// Empty TestPipelineRunName field so that it triggers creation of a new PipelineRun for this Scenario
+		detail.TestPipelineRunName = ""
```

Comment:
I don't think we should erase pipeline here, it reports current status, logic to decide if cancelled pipeline should be replaced must be somewhere else.

Reply:
We agreed to not erase the pipelineRun's names in case they get "Deleted".
Instead, we would simply mark the Snapshot as failed and not re-trigger a new Intg PLR on its behalf.